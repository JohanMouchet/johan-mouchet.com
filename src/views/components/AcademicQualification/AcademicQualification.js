// @flow

import * as React from "react";
import cx from "classnames";
import "./___.scss";

/**
 * Academic Qualifications
 *
 * @param {array}  qualifications 		                 - List of qualifications
 * @param {string} qualifications[0].establishment.name - Qualification establishment name
 * @param {string} qualifications[0].establishment.URL  - Qualification establishment URL
 * @param {string} qualifications[0].location           - Qualification location
 * @param {int}    qualifications[0].graduationYear     - Qualification graduationYear
 * @param {string} qualifications[0].title              - Qualification title
 */

const ___ = (props: Props) => {
  const {
    prop1,
    prop2
  } = props;

  const ___Class = cx(
    "___",
    {
      prop1: "___--prop1"
    }
  );

  return (
	{% for qualification in props.qualifications %}
		<section className="c-academic-qualification">
			<h3 className="c-academic-qualification__heading">
				<a className="c-academic-qualification__establishment" href="{ qualification.establishment.URL }">{ qualification.establishment.name }</a>, { qualification.location }
			</h3>
			<span className="c-academic-qualification__graduation-year">{ qualification.graduationYear }</span>
			<p className="c-academic-qualification__title">{ qualification.title }</p>
		</section>
	{% endfor %}
  );
};

___.defaultProps = {
  prop1: 2
};

export default ___;
