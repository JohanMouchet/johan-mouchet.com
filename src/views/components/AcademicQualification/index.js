export { default } from "./AcademicQualification.js";
