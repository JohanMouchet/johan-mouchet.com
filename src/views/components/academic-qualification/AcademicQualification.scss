@import "assets/styles/settings/_settings";
@import "assets/styles/tools/_tools";

.c-academic-qualification {
  &:not(:last-child) {
    margin-bottom: t-vertical-rhythm(1);
  }

  &__heading {
    margin-bottom: 0;
  }

  &__establishment {
    font-weight: 700;
  }

  &__graduation-year {
    color: $s-color-text-dimmed;
  }
}
