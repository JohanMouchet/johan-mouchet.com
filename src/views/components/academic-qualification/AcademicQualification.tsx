import clsx, { ClassValue } from "clsx";
import styles from "./AcademicQualification.module.scss";

export const AcademicQualification = ({
  degrees,
  className,
  ...props
}: {
  degrees: {
    establishmentName: string;
    establishmentLink: {
      url: string;
    };
    location: string;
    graduationYear: number;
    title: string;
  }[];
  className?: ClassValue;
} & React.HTMLProps<HTMLDivElement>) => {
  if (!degrees?.length) {
    return null;
  }

  return (
    <div
      className={clsx(styles["c-academic-qualifications"], className)}
      {...props}
    >
      {degrees.map((degree) => (
        <section
          className={styles["c-academic-qualification"]}
          key={degree.title}
        >
          <h3 className={styles["c-academic-qualification__heading"]}>
            <a
              className={styles["c-academic-qualification__establishment"]}
              href={degree.establishmentLink.url}
            >
              {degree.establishmentName}
            </a>
            , {degree.location} ⁠—{" "}
            <span className={styles["c-academic-qualification__title"]}>
              {degree.title}
            </span>
          </h3>
          <time
            className={styles["c-academic-qualification__graduation-year"]}
            dateTime={`${degree.graduationYear}`}
          >
            {degree.graduationYear}
          </time>
        </section>
      ))}
    </div>
  );
};
