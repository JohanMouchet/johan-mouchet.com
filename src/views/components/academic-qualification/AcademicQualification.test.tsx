import React from "react";
import { shallow } from "enzyme";
import AcademicQualification from ".";

it("Renders <AcademicQualification /> without crashing", () => {
  expect(
    shallow(
      <AcademicQualification
        degrees={[
          {
            establishment: {
              name: "Name",
              url: "#",
            },
            location: "Location",
            graduationYear: new Date().getFullYear(),
            title: "Title",
          },
        ]}
      />
    )
  ).toMatchSnapshot();
});
