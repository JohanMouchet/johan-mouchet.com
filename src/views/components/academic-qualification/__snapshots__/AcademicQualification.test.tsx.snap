// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <AcademicQualification /> without crashing 1`] = `
<Fragment>
  <section
    className="c-academic-qualification"
    key="Title"
  >
    <h3
      className="c-academic-qualification__heading"
    >
      <a
        className="c-academic-qualification__establishment"
        href="#"
      >
        Name
      </a>
      , 
      Location
    </h3>
    <span
      className="c-academic-qualification__graduation-year"
    >
      2020
    </span>
    <p
      className="c-academic-qualification__title"
    >
      Title
    </p>
  </section>
</Fragment>
`;
